### ISSUE TITLE
Segmentation fault when using :add()

### ISSUE THREAD

**Issue by OsamaSultan** [torch/torch7#336]
Hello,

I'm new to using torch. I was trying to normalise an image dataset as described in the Supervised Learning tutorial. After converting the images to YUV space and calculating their global mean and STD, I then try to subtract the mean from batches of images using

imagesTensor[{ {},j,{},{} }]:add(-mu[j]) 

it goes fine for the testing dataset, but for some reason at the V channel of the second batch of the training dataset this line yields: 

Segmentation fault: 11

P.S. I'm using Torch 7 on MacOSX 10.10

**Comment by fmassa** [comment]
could you provide a minimal example that reproduces your problem ? I can't reproduce your error.

**Comment by OsamaSultan** [comment]
@fmassa Thanks for replying
I finally solved it. It was an bug in my code. I was misusing storages for memory mapping.
